azurelinuxagent/ga/exthandlers.py

```diff
@@ -223,7 +223,15 @@ def run_status(self):
         return
 
     def get_guid(self, name):
-        return self.last_guids.get(name, None)
+        return self.last_guids.get(name, (None, False))[0]
```

Comment:
should you check for None before reference? also the method below

Reply:
`last_upgrade_guids` is initialized to an empty dictionary when the ExtHandlerHandler is initialized.
The second parameter to `get` specifies what is returned if `name` is not found in the dictionary.